dcicutils/sheet_utils.py

```diff
@@ -0,0 +1,218 @@
+import copy
+
+from dcicutils.common import AnyJsonData
+from openpyxl import load_workbook
+from openpyxl.worksheet.worksheet import Worksheet
+from openpyxl.workbook.workbook import Workbook
+from typing import Any, Dict, List, Optional, Union
+
+
+Header = str
+Headers = List[str]
+ParsedHeader = List[Union[str, int]]
+ParsedHeaders = List[ParsedHeader]
+SheetCellValue = Union[int, float, str]
+
+
+class WorkbookManager:
+
+    @classmethod
+    def load_workbook(cls, filename: str):
+        wb = cls(filename)
+        return wb.load_content()
+
+    def __init__(self, filename: str):
+        self.filename: str = filename
+        self.workbook: Optional[Workbook] = None
+        self.headers_by_sheetname: Dict[str, List[str]] = {}
+        self.content_by_sheetname: Dict[str, List[Any]] = {}
+
+    def sheet_headers(self, sheetname: str) -> List[str]:
+        return self.headers_by_sheetname[sheetname]
+
+    def sheet_content(self, sheetname: str) -> List[Any]:
+        return self.content_by_sheetname[sheetname]
+
+    @classmethod
+    def _all_rows(cls, sheet: Worksheet):
+        row_max = sheet.max_row
+        for row in range(2, row_max + 1):
+            yield row
+
+    @classmethod
+    def _all_cols(cls, sheet: Worksheet):
+        col_max = sheet.max_column
+        for col in range(1, col_max + 1):
+            yield col
```

Comment:
Not to belabor the point as we've discussed this before, but I don't really understand the use of `@classmethod` here when there are no references to `cls`. I looked this up and as it turns out most styling standards do not really mind that you do this, so I guess it's ok, but generally I find it confusing as when I see a `@classmethod` I am implicitly assuming there is a dependency for that method elsewhere in the class ie: it cannot be ported as a standalone function outside of the class.

Reply:
I do not like @staticmethod because it's not a concept in many languages. A compiler can figure out what's a static method and there should be no need to declare it. If I want to call other class methods, I like having `cls` already available as a variable so I don't have to distract myself by adding data flow to receive the class name. If something is truly promised to be a static method, I just make it a function. There is no need to go through the syntax of calling it indirect through a class. It could be called by any class and it's weird to hide it somewhere that makes it syntactically hard for other things to get to it.  If it's part of a class, then it should have access to the class methods.

Also, as a practical matter, Python fusses if something is an instance method but could be a static method but it does not fuss if it's a class method and could be a static method, probably because it wants to accommodate exactly the programming style that I use.